discovery/gossiper.go

```diff
@@ -511,7 +511,7 @@ func (d *AuthenticatedGossiper) retransmitStaleChannels() error {
 		// If it's been a full day since we've re-broadcasted the
 		// channel, add the channel to the set of edges we need to
 		// update.
-		if timeElapsed >= broadcastInterval {
+		if timeElapsed >= broadcastInterval && info.AuthProof != nil {
```

Comment:
Minor nit: the above comment should also be updated to outline that we check the auth proof, as we shouldn't broadcast channels which we don't have the full proofs for, or have not be _announced_ to the network. In the case of the latter case, it may be a privacy leak.